src/authn/ccloudProvider.ts

```diff
@@ -446,8 +436,8 @@ export class ConfluentCloudAuthProvider
 
     // general listener for the URI handling event, which is used to resolve any auth flow promises
     // and will trigger the secrets.onDidChange event described above
-    const uriHandlerSub: vscode.Disposable = getUriHandler().event(
-      async (uri) => await this.handleUri(uri),
+    const ccloudAuthCallbackSub: vscode.Disposable = ccloudAuthCallback.event(
```

Comment:
These event handlers should be factored out to individual methods, then have the .bind(this) of each be registered as event handlers in future cleanup.

This one is already nearly that way, suggest:

    const ccloudAuthCallbackSub: vscode.Disposable = ccloudAuthCallback.event(
      this.handleCCloudAuthCallback.bind(this),
    );

Reply:
Ah good point - I'll want to do the same for the secrets change listener as well, tracking in https://github.com/confluentinc/vscode/issues/2734 for a follow-up branch that will handle updating those three listeners